Lib/posixpath.py

```diff
@@ -329,61 +308,55 @@ def expandvars(path):
             break
         i, j = m.span(0)
         name = m.group(1)
-        if name.startswith(start) and name.endswith(end):
+        if name[:1] == start and name[-1:] == end:
             name = name[1:-1]
         try:
             if environ is None:
                 value = os.fsencode(os.environ[os.fsdecode(name)])
             else:
                 value = environ[name]
-        except KeyError:
-            i = j
-        else:
             tail = path[j:]
             path = path[:i] + value
             i = len(path)
             path += tail
+        except KeyError:
```

Comment:
Does this speed anything up or fix any bugs?

Reply:
No.

Reply:
Then this change should not be included in this PR